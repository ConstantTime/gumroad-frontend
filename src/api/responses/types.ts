export interface IProduct {
  productId: String;
  name: String;
  description: String;
}

export interface IRating {
  ratingId: String;
  rating: number;
  review: String;
}
