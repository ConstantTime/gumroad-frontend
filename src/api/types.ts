export interface IFetchDataParams {
  endpoint: string;
  method: EHttpMethod;
  body?: object;
  payloadToQueryParam?: boolean;
}

export enum EHttpMethod {
  GET = "GET",
  POST = "POST",
  PUT = "PUT",
  PATCH = "PATCH",
  DELETE = "DELETE",
}

export interface IInitOptions {
  method: EHttpMethod;
  body?: {};
}
