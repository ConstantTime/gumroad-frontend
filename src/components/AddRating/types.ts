export interface IAddRatingModal {
  closeModal: VoidFunction;
}
