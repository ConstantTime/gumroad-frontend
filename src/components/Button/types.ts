export interface IGumroadButton {
  onClick: VoidFunction;
}
