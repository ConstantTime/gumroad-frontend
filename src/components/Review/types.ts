import { IRating } from "../../api/responses/types";

export interface IReview {
  rating: IRating;
}
