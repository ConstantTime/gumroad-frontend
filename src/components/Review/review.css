.review {
  display: flex;
  align-items: center;
}

.rating {
  margin-left: 10px;
  margin-right: 5px;
}

.review-text {
  color: gray;
}
