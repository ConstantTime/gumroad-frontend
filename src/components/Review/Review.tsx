import Rating from "@material-ui/lab/Rating";
import React from "react";
import "./review.css";
import { IReview } from "./types";

const Review: React.FC<IReview> = ({ rating }) => {
  return (
    <div className="review">
      <Rating
        name="half-rating"
        value={rating.rating}
        precision={0.5}
        readOnly
      />
      <div className="rating">{rating.rating}</div>
      <div className="review-text">{rating.review}</div>
    </div>
  );
};

export default Review;
