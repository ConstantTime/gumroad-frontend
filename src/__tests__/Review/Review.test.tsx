import { shallow } from "enzyme";
import { IRating } from "../../api/responses/types";
import Review from "../../components/Review/Review";

describe("Review", () => {
  it("should render the component", () => {
    const props: IRating = {
      ratingId: "1",
      rating: 2.5,
      review: "hi",
    };
    const wrapper = shallow(<Review rating={props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
