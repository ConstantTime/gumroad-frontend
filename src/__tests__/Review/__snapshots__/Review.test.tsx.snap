// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should render the component 1`] = `
<div
  className="review"
>
  <WithStyles(ForwardRef(Rating))
    name="rating-stars"
    precision={0.5}
    readOnly={true}
    value={2.5}
  />
  <div
    className="rating"
  >
    2.5
  </div>
  <div
    className="review-text"
  >
    hi
  </div>
</div>
`;
